src/plugins/core/conditional_format.ts

```diff
@@ -88,6 +88,57 @@ export class ConditionalFormatPlugin
             );
             break;
         }
+      } else if (rule.rule.type === "CellIsRule") {
```

Comment:
Doesn't work for formulas referencing a range on another sheet (`=SomeOtherSheet!A1`). You should loop on all the CFs, and not on only the CFs onf the sheet that has had the range change.

Same for data validation.

Data bars also kinda have the issue, but entering a sheet on another range is totally broken. That can probably be fixed [here](https://www.odoo.com/odoo/2328/tasks/4072640)